go/test/endtoend/schemadiff/vrepl/schemadiff_vrepl_suite_test.go

```diff
@@ -0,0 +1,427 @@
+		row := rs.Named().Row()
+		require.NotNil(t, row)
+		mysqlVersion := row["ver"].ToString()
+
+		if re.MatchString(mysqlVersion) {
+			t.Skipf("Skipping test due to ignore_versions=%s", ignoreVersions)
+			return
+		}
+	}
+	if _, exists := readTestFile(t, testName, "expect_query_failure"); exists {
+		// VTGate failure is expected!
+		// irrelevant to this suite.
+		// We only want to test actual migrations
+		t.Skip("expect_query_failure found. Irrelevant to this suite")
+		return
+	}
+	if _, exists := readTestFile(t, testName, "expect_failure"); exists {
+		// irrelevant to this suite.
+		// We only want to test actual migrations
+		t.Skip("expect_failure found. Irrelevant to this suite")
+		return
+	}
+
+	sqlModeQuery := fmt.Sprintf("set @@global.sql_mode='%s'", defaultSQLMode)
+	_ = mysqlExec(t, sqlModeQuery, "")
+	_ = mysqlExec(t, "set @@global.event_scheduler=0", "")
+
+	_ = mysqlExec(t, fmt.Sprintf("drop table if exists %s", tableName), "")
+	_ = mysqlExec(t, fmt.Sprintf("drop event if exists %s", eventName), "")
+
+	var fromCreateTable string
+	var toCreateTable string
+	{
+		// create
+		f := "create.sql"
+		_, exists := readTestFile(t, testName, f)
+		require.True(t, exists)
+		onlineddl.MysqlClientExecFile(t, mysqlParams(), testDataPath, testName, f)
+		// ensure test table has been created:
+		// read the create statement
+		fromCreateTable = getCreateTableStatement(t, tableName)
+		require.NotEmpty(t, fromCreateTable)
+	}
+	defer func() {
+		// destroy
+		f := "destroy.sql"
+		if _, exists := readTestFile(t, testName, f); exists {
+			onlineddl.MysqlClientExecFile(t, mysqlParams(), testDataPath, testName, f)
+		}
+	}()
+
+	// Run test
+	alterClause := "engine=innodb"
+	if content, exists := readTestFile(t, testName, "alter"); exists {
+		alterClause = content
+	}
+	alterStatement := fmt.Sprintf("alter table %s %s", tableName, alterClause)
+	// Run the DDL!
+	onlineddl.VtgateExecQuery(t, &vtParams, alterStatement, "")
+	// migration is complete
+	// read the table structure of modified table:
+	toCreateTable = getCreateTableStatement(t, tableName)
+	require.NotEmpty(t, toCreateTable)
+
+	if content, exists := readTestFile(t, testName, "expect_table_structure"); exists {
+		assert.Contains(t, toCreateTable, content, "expected SHOW CREATE TABLE to contain text in 'expect_table_structure' file")
+	}
+
+	fromTestTableSchemas = append(fromTestTableSchemas, &testTableSchema{
+		testName:    testName,
+		tableSchema: fromCreateTable,
+	})
+	toTestTableSchemas = append(fromTestTableSchemas, &testTableSchema{
+		testName:    testName,
+		tableSchema: toCreateTable,
+	})
+
+	hints := &schemadiff.DiffHints{}
+	if strings.Contains(alterClause, "AUTO_INCREMENT") {
+		hints.AutoIncrementStrategy = schemadiff.AutoIncrementApplyAlways
+	}
+	t.Run("validate diff", func(t *testing.T) {
+		validateDiff(t, fromCreateTable, toCreateTable, hints)
+	})
+}
+
+// func TestRandomSchemaChanges(t *testing.T) {
```

Comment:
did you intentionally leave this around?

Reply:
Actually, yes. So I had this idea that on top of the existing 80 - 90 tests, I could also do a Cartesian product of all tests, a crossover between them all, to test any-schema-to-any-schema conversion. That's the commented code. The reason it's commented is that this leads to some 5K tests, which run well within 7 minutes -- a reasonable time  -- but I'm afraid that if I add a bit more test, this will explode (runtime is n^2).